tests/service/test_dev.py

```diff
@@ -371,3 +370,66 @@ async def test_dep_check(self):
             Install.dep_check(
                 {("model", "vowpalWabbit"): {"feedface": lambda: False}}, []
             )
+
+
+class TestMakeDocs(AsyncTestCase):
+
+    root = pathlib.Path(__file__).parents[2]
+    docs_root = root / "docs"
+
+    symlinks_to_chk = [
+        (("changelog.md",), ("CHANGELOG.md",)),
+        (("shouldi.md",), ("examples", "shouldi", "README.md",)),
+        (("swportal.rst",), ("examples", "swportal", "README.rst",)),
+        (
+            ("contributing", "consoletest.md",),
+            ("dffml", "util", "testing", "consoletest", "README.md",),
+        ),
+        (("plugins", "service", "http",), ("service", "http", "docs",),),
+    ]
+
+    files_to_check = [
+        (file_name,) for file_name in os.listdir(docs_root / "images")
+    ] + [("_static", "copybutton.js",), (".nojekyll",)]
+
+    async def test_files(self):
+        if not is_develop("dffml"):
+            self.skipTest("dffml not installed in development mode")
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            with unittest.mock.patch(
+                "asyncio.create_subprocess_exec", new=mkexec()
+            ):
+                await Develop.cli("docs", "-target-dir", tempdir)
+
+                for symlink, source in self.symlinks_to_chk:
+                    symlink_path = self.docs_root.joinpath(*symlink)
+                    source_path = self.root.joinpath(*source)
+                    self.assertTrue(symlink_path.exists())
+                    self.assertTrue(symlink_path.resolve() == source_path)
+
+                for file_name in self.files_to_check:
+                    file_path = pathlib.Path(tempdir).joinpath(*file_name)
+                    self.assertTrue(file_path.exists())
+
+    async def test_cmd_seq(self):
+        if not is_develop("dffml"):
+            self.skipTest("dffml not installed in development mode")
+
+        stdout = io.StringIO()
+
+        with unittest.mock.patch(
+            "asyncio.create_subprocess_exec", new=mkexec()
+        ), contextlib.redirect_stdout(stdout):
+            with tempfile.TemporaryDirectory() as tempdir:
+                await Develop.cli("docs", "-target-dir", tempdir)
+                self.assertEqual(
```

Comment:
We want to keep as few things in `redirect_stdout/stderr` blocks as possible. Just incase future people modify within the block and wonder why they aren't getting output.

Reply:
done